packages/beacon-node/src/chain/archiveStore/differentialState/metrics.ts

```diff
@@ -0,0 +1,89 @@
+import {Gauge, Histogram} from "@lodestar/utils";
+import {RegistryMetricCreator} from "../../../metrics/index.ts";
+import {RegenErrorType} from "../historicalState/types.ts";
+
+export type DifferentialStateRegenMetrics = {
+  regenTime: Histogram;
+  blockReplayTime: Histogram;
+  blockReplayCount: Histogram;
+  regenRequestCount: Gauge;
+  regenSuccessCount: Gauge;
+  regenErrorCount: Gauge<{reason: RegenErrorType}>;
+  loadSnapshotStateTime: Histogram;
+  loadDiffStateTime: Histogram;
+  computeDiffStateTime: Histogram;
+  applyDiffStateTime: Histogram;
+  stateDiffSize: Gauge;
+  stateSnapshotSize: Gauge;
+};
+
+export function createDifferentialStateRegenMetrics(
+  metricsRegister: RegistryMetricCreator
+): DifferentialStateRegenMetrics {
+  return {
+    regenTime: metricsRegister.histogram({
+      name: "lodestar_differential_state_regen_time_seconds",
+      help: "Time to regenerate a differential state in seconds",
+      // Historical state regen can take up to 3h as of Aug 2024
+      // 5m, 10m, 30m, 1h, 3h
+      buckets: [5 * 60, 10 * 60, 30 * 60, 60 * 60, 180 * 60],
+    }),
+    blockReplayTime: metricsRegister.histogram({
+      name: "lodestar_differential_state_block_replay_time_seconds",
+      help: "Time to run block replay to regen differential state in seconds",
+      // 5m, 10m, 30m, 1h, 3h
+      buckets: [5 * 60, 10 * 60, 30 * 60, 60 * 60, 180 * 60],
+    }),
+    blockReplayCount: metricsRegister.histogram({
+      name: "lodestar_differential_state_block_replay_count",
+      help: "Count of blocks processed during state transition to differential historical state",
```

Comment:
Note that this is count of slots not count of blocks. There could be empty slots